encfs/BlockFileIO.cpp

```diff
@@ -171,11 +171,11 @@ ssize_t BlockFileIO::read(const IORequest &req) const {
   return result;
 }
 
-bool BlockFileIO::write(const IORequest &req) {
+int BlockFileIO::write(const IORequest &req) {
   CHECK(_blockSize != 0);
 
   off_t fileSize = getSize();
-  if (fileSize < 0) return false;
+  if (fileSize < 0) return fileSize;
```

Comment:
Thank you for your review @samrocketman 👍 
So with this change, if BlockFileIO::write return value `<= 0`, it means it failed.
If return value `== 0`, it means it succeeded.
This function does not return value `> 0`.
So line 256 will break only if write failed, and the error code will be properly returned 👍 
Generally in C, return code `0` means it succeeded :)